### PR TITLE
Add basic functional tests

### PR DESCRIPTION
**Pull request by kant2002**
- Compare results of CL.exe compilation with cesium.

Not sure if this test framework will fly or not, and it is not working on Linux yet. 
Based on my understanding how generators should be tested this is best long-term approach.

### CODE REVIEW

**Comment by ForNeVeR** [comment]
I'm okay with the idea in general, but will make some changes in the script and wire it up with the CI.

**Comment by kant2002** [comment]
I would like to hear more. I work from Windows and Linux so this is at least half of the work. Also I would like tot use this for testing basic runtime library.

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,4 @@
 bin/
 obj/
 
-*.dll
```
Comment (kant2002) [inline .gitignore:7]: Can you leave this? If you run commands from readme it pollutes `git status`

Comment (ForNeVeR) [reply .gitignore:7]: Yeah, sure.

**Review by ForNeVeR — APPROVED** [review]
Thanks for help! Now this will work in CI.

**Pull request merged**